source/time_tables/get_stop_times.cpp

```diff
@@ -59,23 +65,23 @@ std::vector<datetime_stop_time> get_stop_times(const std::vector<type::idx_t>& j
             if(!jpp->stop_point->accessible(accessibilite_params.properties)) {
                 continue;
             }
-            auto st = next_st.earliest_stop_time(routing::JppIdx(*jpp),
-                                                 next_requested_datetime[jpp_idx],
-                                                 disruption_active,
-                                                 accessibilite_params.vehicle_properties);
+            auto st = next_stop_times(routing::JppIdx(*jpp), next_requested_datetime[jpp_idx]);
 
             if (st.first != nullptr) {
                 DateTime dt_temp = st.second;
-                if (dt_temp <= max_dt) {
+                if ( (clockwise && dt_temp <= max_dt) || (!clockwise && dt_temp >= max_dt)) {
                     result.push_back(std::make_pair(dt_temp, st.first));
                     test_add = true;
                     // The next stop time must be at least one second after
-                    next_requested_datetime[jpp_idx] = dt_temp + 1;
+                    next_requested_datetime[jpp_idx] = clockwise ? dt_temp + 1 : dt_temp - 1;
```

Comment:
I prefere `next_requested_datetime[jpp_idx] = dt_temp + (clockwise ? 1 : -1);`, but that's just personal preference.